File: hw2/prime_numbers.py
import math
from datetime import datetime


def is_prime(number: int) -> bool:
    if number < 2:
        return False
    for i in range(2, int((math.sqrt(number)) + 1)):
        if number % i == 0:
            return False
    return True


def get_n_primes(number: int) -> list:
    list_of_primes = []
    i = 2

    while len(list_of_primes) < number:
        if is_prime(i):
            list_of_primes.append(i)
        i += 1

    return list_of_primes


if __name__ == "__main__":
    start_time = datetime.now()
    print(get_n_primes(100))
    print(f'Время выполнения {datetime.now() - start_time}')

File: hw2/prime_numbers.js
function getNprimes(n){
  const arr = [];
  let i = 2

  while (arr.length < n) {
    if (isPrime(i)) {
      arr.push(i)
    }
    i++
  } 
  return arr;

  function isPrime(n) {

    if ( n < 2 ) {
      return false
    }

    for ( let i = 2; i <= Math.sqrt(n); i++ ) {
      if ( n % i === 0 ) {
          return false;
      } 
    }
    return true
  }

}

console.time();
console.log(getNprimes(process.argv[2]));
console.timeEnd();

